src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignmentRegion.java

```diff
@@ -0,0 +1,169 @@
+        this.forwardStrand = ! read.isReverseStrand();
+        this.cigar = forwardStrand ? read.getCigar() : CigarUtils.invertCigar(read.getCigar());
+        this.referenceInterval = new SimpleInterval(read);
+        this.assembledContigLength = cigar.getReadLength() + getHardClipping(cigar);
+        this.startInAssembledContig = startOfAlignmentInContig(cigar);
+        this.endInAssembledContig = endOfAlignmentInContig(assembledContigLength, cigar);
+        this.mqual = read.getMappingQuality();
+        if (read.hasAttribute("NM")) {
+            this.mismatches = read.getAttributeAsInteger("NM");
+        } else {
+            this.mismatches = 0;
+        }
+    }
+
+    public int overlapOnContig(final AlignmentRegion other) {
+        return Math.max(0, Math.min(endInAssembledContig, other.endInAssembledContig) - Math.max(startInAssembledContig, other.startInAssembledContig));
+    }
+
+    private static int getHardClipping(final Cigar cigar) {
+        final List<CigarElement> cigarElements = cigar.getCigarElements();
+        return (cigarElements.get(0).getOperator() == CigarOperator.HARD_CLIP ? cigarElements.get(0).getLength() : 0) +
+                (cigarElements.get(cigarElements.size() - 1).getOperator() == CigarOperator.HARD_CLIP ? cigarElements.get(cigarElements.size() - 1).getLength() : 0);
+    }
+
+    private static int startOfAlignmentInContig(final Cigar cigar) {
+        return getClippedBases(true, cigar) + 1;
+    }
+
+    private static int endOfAlignmentInContig(final int assembledContigLength, final Cigar cigar) {
+        return assembledContigLength - getClippedBases(false, cigar);
+    }
+
+    private static int getClippedBases(final boolean fromStart, final Cigar cigar) {
+        int posInContig = 0;
+        int j = fromStart ? 0 : cigar.getCigarElements().size() - 1;
+        final int offset = fromStart ? 1 : -1;
+        CigarElement ce = cigar.getCigarElement(j);
+        while (ce.getOperator().isClipping()) {
+            posInContig += ce.getLength();
+            j += offset;
+            ce = cigar.getCigarElement(j);
+        }
+        return posInContig;
+    }
+
+    @Override
+    public String toString() {
+        return breakpointId +
+                "\t" +
+                contigId +
+                "\t" +
+                referenceInterval.getContig() +
+                "\t" +
+                referenceInterval.getStart() +
+                "\t" +
+                referenceInterval.getEnd() +
+                "\t" +
+                (forwardStrand ? "+" : "-") +
+                "\t" +
+                cigar.toString() +
+                "\t" +
+                mqual +
+                "\t" +
+                startInAssembledContig +
+                "\t" +
+                endInAssembledContig +
+                "\t" +
+                mismatches;
+    }
+
+    public static AlignmentRegion fromString(final String[] fields) {
+        final String breakpointId = fields[0];
```

Comment:
Because this is a public method, it makes sense to document what is the expected order in the input array.